File: ui/src/app/model/projeto-fase.ts
import { FormArray, FormControl, FormGroup, Validators } from '@angular/forms';

import { Projeto } from './projeto';
import { ProjetoFaseResponsavel } from './projeto-fase-responsavel';
import { StatusFase } from './status-fase';

export class ProjetoFase {

  id: number | undefined;
  codigo: number | undefined;
  descricao: string | undefined;
  dataInicio: Date | undefined;
  dataFim: Date | undefined;
  projeto: Projeto | undefined;
  andamento: number | undefined;
  levantamentoRequisito: string | undefined;
  statusFase: StatusFase | undefined;
  custoPrevisto: number | undefined;
  custoExecutado: number | undefined;
  projetoFaseResponsaveis: ProjetoFaseResponsavel [] | undefined;

  constructor() {
    this.projetoFaseResponsaveis = [];
    this.andamento = 0;
    this.custoExecutado = 0;
    this.custoPrevisto = 0;
  }

  criarFormulario(projetoFase: ProjetoFase) {

    return new FormGroup({
      id: new FormControl(projetoFase.id),
      codigo: new FormControl(projetoFase.codigo, Validators.required),
      descricao: new FormControl(projetoFase.descricao, Validators.required),
      dataInicio: new FormControl(projetoFase.dataInicio, Validators.required),
      dataFim: new FormControl(projetoFase.dataFim, Validators.required),
      projeto: new FormControl(projetoFase.projeto, Validators.required),    
      andamento: new FormControl(projetoFase.andamento),
      levantamentoRequisito: new FormControl(projetoFase.levantamentoRequisito),
      statusFase: new FormControl(projetoFase.statusFase),
      custoPrevisto: new FormControl(projetoFase.custoPrevisto),
      custoExecutado: new FormControl(projetoFase.custoExecutado),
      projetoFaseResponsaveis: new FormArray(projetoFase!.projetoFaseResponsaveis!.map(responsavel => ProjetoFaseResponsavel.criarFormulario(responsavel)))
    });
  }
  
}

File: api/src/main/java/com/example/demo/model/ProjetoFase.java
package com.example.demo.model;

import com.example.demo.model.enuns.StatusFase;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "projeto_fase")
public class ProjetoFase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "codigo")
    private Integer codigo;

    @Column(name = "descricao")
    private String descricao;

    @Column(name = "data_inicio")
    private LocalDate dataInicio;

    @Column(name = "data_fim")
    private LocalDate dataFim;

    @ManyToOne
    @JoinColumn(name = "projeto_id")
    private Projeto projeto;

    @Column(name = "andamento")
    private Double andamento;

    @Column(name = "levantamento_requisito")
    private String levantamentoRequisito;

    @Enumerated(EnumType.STRING)
    @Column(name = "status_fase")
    private StatusFase statusFase;

    @Column(name = "custo_previsto")
    private BigDecimal custoPrevisto;

    @Column(name = "custo_executado")
    private BigDecimal custoExecutado;

    @ElementCollection
    @CollectionTable(name = "projeto_fase_responsavel", joinColumns = {@JoinColumn(name = "projeto_fase_id")})
    private List<ProjetoFaseResponsavel> projetoFaseResponsaveis = new ArrayList<>();

}
